### ISSUE TITLE
Support Corepack

### ISSUE THREAD

**Issue by AlexWayfer** [nodenv/nodenv#181]
Node.js 16.9 including Corepack: https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V16.md#2021-09-07-version-1690-current-targos

[Its documentation says](https://github.com/nodejs/corepack#usage) that I just can use `pnpm`, but I can't (with `nodenv`): 

```
> nodenv install
Downloading node-v16.9.1-linux-x64.tar.gz...
-> https://nodejs.org/dist/v16.9.1/node-v16.9.1-linux-x64.tar.gz
Installing node-v16.9.1-linux-x64...
Installed node-v16.9.1-linux-x64 to /home/alex/.nodenv/versions/16.9.1

> pnpm i
nodenv: pnpm: command not found

The `pnpm' command exists in these Node versions:
  16.6.1
  16.7.0
```

**Comment by Ayc0** [comment]
I have a similar issue with running: `corepack enable`: I have this error: `nodenv: corepack: command not found`

**Comment by Pyrolistical** [comment]
looks like the corepack shim was added

**Comment by Pyrolistical** [comment]
Odd. On another machine the phnom shim is missing. The one that works is a x86 and the one that doesn't is arm64

I wonder if there is simply a bundling error

**Comment by 0xdevalias** [comment]
> looks like the corepack shim was added

I'm using `1.4.0` and while I don't get an error while calling `corepack`, it doesn't actually seem to create any shims/etc that allow it to hijack `npm`/`yarn`/etc.

I can get around it by running `corepack yarn --version`, but that isn't ideal.

```
⇒  which -a yarn
/Users/devalias/.nodenv/shims/yarn
/usr/local/bin/yarn

⇒  which -a corepack
/Users/devalias/.nodenv/shims/corepack

⇒  nodenv --version
nodenv 1.4.0

⇒  yarn --version
1.22.19

⇒  corepack yarn --version
1.22.5

⇒  /Users/devalias/.nodenv/shims/yarn --version
1.22.5

⇒  /usr/local/bin/yarn --version
1.22.19
```

**Comment by jasonkarns** [comment]
I haven't had much time recently, but will be looking into this soon.

I'm going to need to dive into the documentation but it seems that on a fresh node install, one needs to `corepack enable`, yes? Which creates shims for `yarn` and `pnpm` (only?).

So the presumption is that a) yarn and pnpm _themselves_ are uninstalled first. If they exist, nodenv's shims will hit before getting to corepack's shims for them.

but more importantly, b) yarn and pnpm would need to be uninstalled _across all nodes_ because nodenv's shim for yarn/pnpm exist if _any_ node has them installed globally. (This is intentional! It is those very shims that must be invoked in order to direct the command to the appropriate node based on env-var/.node-version/etc.)

IOW, this might be tricky.

**Comment by yshrsmz** [comment]
Yes, `corepack enable` (and explicit `corepack enable npm` for npm) is needed IIUC(and is still in the experimental stage).
https://nodejs.org/api/corepack.html#enabling-the-feature

nodenv installs each version separately, so personally, I think it's kind of natural that one needs to call `corepack enable` for each instance of node.js.

**Comment by 0xdevalias** [comment]
> nodenv installs each version separately, so personally, I think it's kind of natural that one needs to call corepack enable for each instance of node.js

That would also be how I would expect the behaviour to be.
